test/parallel/test-stream2-readable-bytesread.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+require('../common');
+var assert = require('assert');
+
+var Readable = require('stream').Readable;
+var Duplex = require('stream').Duplex;
+var Transform = require('stream').Transform;
+
+(function() {
+  var readable = new Readable({
+    read: function(n) {
+      var i = this._index++;
+      if (i > this._max)
+        this.push(null);
+      else {
+        this.push(new Buffer('a'));
```

Comment:
@mscdex The `instanceof` check has been removed. Buffer doesn't care whether you use `new` or not. It always returns of a new `Uint8Array` instance regardless.

@indutny Can see what you're saying. Unfortunate that it's a byproduct of using typed arrays. We can continue to use `new`. Not going to break anything. Just adds an extra object allocation to the call.